File: Array/1.Easy/08.Linear_search.cpp
/*
I don't think anyone needs it's solution. The idea is to traverse the array using loop and when the element
is equal to k return the same
*/

#include <bits/stdc++.h>
using namespace std;

// CODE:-
int linearSearch(int arr[], int n, int k)
{
    for (int i = 0; i < n; i++)
    {
        if (arr[i] == k)
            return i;
    }
    return -1;
}

int main()
{
    int arr[] = {5, 3, 7, 2, 8, 1, 9};
    int n = 7;
    int k = 8;
    
    cout << "Array: ";
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;
    
    int index = linearSearch(arr, n, k);
    
    if (index != -1)
        cout << "Element " << k << " found at index: " << index << endl;
    else
        cout << "Element " << k << " not found" << endl;
    
    return 0;
}

// TIME COMPLEXITY = O(N)

File: 01.Arrays/1.Easy/08.Linear_search.py
# Traverse the array using loop and when the element is equal to k return the index

# CODE:-
def linearSearch(arr, n, k):
    """Find the index of element k in array"""
    for i in range(n):
        if arr[i] == k:
            return i
    return -1


if __name__ == '__main__':
    arr = [5, 3, 7, 2, 8, 1, 9]
    n = 7
    k = 8
    
    print("Array:", arr)
    
    index = linearSearch(arr, n, k)
    
    if index != -1:
        print(f"Element {k} found at index: {index}")
    else:
        print(f"Element {k} not found")

# TIME COMPLEXITY = O(N) 
